py/day1.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+from queue import PriorityQueue
+from typing import Generator
+
+
+def ingest_inventory_sums(input_file: Path) -> Generator[int, None, None]:
+    assert input_file.exists(), f"{input_file} does not exist"
+
+    current_elf: list[int] = []
+
+    with open(input_file) as f_in:
+        for line in f_in:
+            # check for blank line indicating another elf's inventory
+            if line.startswith("\n"):
+                yield sum(current_elf)
```

Comment:
Can you explain your use of yield here?

Reply:
This section of this larger article is a good TLDR for generator functions.
https://realpython.com/introduction-to-python-generators/#understanding-generators

>Generator functions look and act just like regular functions, but with one defining characteristic. Generator functions use the Python [yield keyword](https://realpython.com/python-keywords/#returning-keywords-return-yield) instead of return. Recall the generator function you wrote earlier:

```python
def infinite_sequence():
    num = 0
    while True:
        yield num
        num += 1
```

>This looks like a typical [function definition](https://realpython.com/defining-your-own-python-function/), except for the Python `yield` statement and the code that follows it. yield indicates where a value is sent back to the caller, but unlike return, you don’t exit the function afterward.

>Instead, the **state** of the function is remembered. That way, when `next()` is called on a generator object (either explicitly or implicitly within a for loop), the previously yielded variable num is incremented, and then yielded again. Since generator functions look like other functions and act very similarly to them, you can assume that generator expressions are very similar to other comprehensions available in Python.